nexus/db-queries/src/db/queries/sled_reservation.rs

```diff
@@ -57,18 +57,70 @@ pub fn sled_find_targets_query(
                 COALESCE(SUM(CAST(sled_resource_vmm.reservoir_ram AS INT8)), 0) + "
             ).param().sql(" <= sled.reservoir_size
         ),
-        our_aa_groups AS (
+        our_a_groups AS (
+            SELECT group_id
+            FROM affinity_group_instance_membership
+            WHERE instance_id = ").param().sql("
+        ),
+        other_a_instances AS (
+            SELECT affinity_group_instance_membership.group_id,instance_id
+            FROM affinity_group_instance_membership
+            JOIN our_a_groups
+            ON affinity_group_instance_membership.group_id = our_a_groups.group_id
+            WHERE instance_id != ").param().sql("
+        ),
+        our_direct_aa_groups AS (
+            -- Anti-affinity groups to which our instance belongs
             SELECT group_id
             FROM anti_affinity_group_instance_membership
             WHERE instance_id = ").param().sql("
         ),
-        other_aa_instances AS (
+        other_direct_aa_instances AS (
             SELECT anti_affinity_group_instance_membership.group_id,instance_id
             FROM anti_affinity_group_instance_membership
-            JOIN our_aa_groups
-            ON anti_affinity_group_instance_membership.group_id = our_aa_groups.group_id
+            JOIN our_direct_aa_groups
+            ON anti_affinity_group_instance_membership.group_id = our_direct_aa_groups.group_id
             WHERE instance_id != ").param().sql("
         ),
+        our_indirect_aa_groups AS (
```

Comment:
The input to this *whole* query is "our instance ID".

We looked up the affinity groups this instance belongs to earlier - that's saved as `our_a_groups`.

So this subquery looks up "memberships in *any* anti-affinity groups, based on **this instance's** affinity groups".

This is "indirect" because our instance may not directly belong to this anti-affinity group, but we're looking it up because our instance needs to consider it through the affinity-group member relationship.

(The `exactly_one_affinity_group` value we return here is used below...)